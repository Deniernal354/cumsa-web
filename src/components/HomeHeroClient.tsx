"use client";
import { motion } from "framer-motion";
import Image from "next/image";
import Link from "next/link";
import Typewriter from "typewriter-effect";

export default function HomeHeroClient() {
  return (
    <section className="relative overflow-hidden">
      <div className="absolute inset-0 -z-10">
        <Image
          src="/zahid-lilani-unsplash-scaled.jpg"
          alt="CUMSA banner"
          fill
          priority
          className="object-cover"
        />
        <div className="absolute inset-0 bg-[var(--navy)]/55" />
      </div>
      <div className="container-base py-24 md:py-32 text-white">
        <motion.h1
          initial={{ opacity: 0, y: 12 }}
          animate={{ opacity: 1, y: 0 }}
          transition={{ duration: 0.5, ease: "easeOut" }}
          className="text-4xl md:text-6xl font-extrabold tracking-tight"
        >
          Carleton University Muslim Studentsâ€™ Association
        </motion.h1>
        <motion.div className="mt-4 max-w-2xl text-white/85 text-lg">
          <Typewriter
            options={{
              autoStart: true,
              loop: true,
              delay: 38,
              deleteSpeed: 22,
              cursor: '<span class="typewriter-cursor">|</span>',
              strings: [
                "Join us for prayer, support, and vibrant community!",
                "Family in Faith on campus.",
                "la ilaha illallah muhammadur rasulullah.",
              ],
            }}
          />
        </motion.div>
        <motion.div
          initial={{ opacity: 0 }}
          animate={{ opacity: 1 }}
          transition={{ delay: 0.2, duration: 0.5 }}
          className="mt-8 flex flex-wrap gap-3"
        >
          <Link
            href="/services/prayer"
            className="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-[var(--blue)] text-white hover:opacity-90 focus-visible:ring-[var(--blue)] ring-offset-[var(--surface)] px-4 py-2"
          >
            Prayers
          </Link>
          <Link
            href="/join-us"
            className="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none border border-black/10 bg-[var(--surface)] text-[var(--foreground)] hover:bg-black/5 focus-visible:ring-[var(--blue)] ring-offset-[var(--surface)] px-4 py-2"
          >
            Join Us
          </Link>
        </motion.div>
      </div>
    </section>
  );
}
