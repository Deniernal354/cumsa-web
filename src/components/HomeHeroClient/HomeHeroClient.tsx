"use client";
import { motion } from "framer-motion";
import Image from "next/image";
import Link from "next/link";
import Typewriter from "typewriter-effect";
import { useEffect, useState, useRef } from "react";
import { FaChevronLeft, FaChevronRight } from "react-icons/fa";

export default function HomeHeroClient() {
  const heroImages = [
    "/hero/MUSULLAH.png",
    "/hero/UC.jpeg",
    "/hero/zahid-lilani-unsplash-scaled.jpg",
  ];
  const [current, setCurrent] = useState(0);

  useEffect(() => {
    const timer = setInterval(() => {
      setCurrent((i) => (i + 1) % heroImages.length);
    }, 6000);
    return () => clearInterval(timer);
  }, [heroImages.length]);

  const sectionRef = useRef<HTMLDivElement>(null);

  type SlideControlsProps = {
    current: number;
    setCurrent: React.Dispatch<React.SetStateAction<number>>;
    heroImages: string[];
  };
  function SlideControls({ current, setCurrent, heroImages }: SlideControlsProps) {
    const total = heroImages.length;
    const prev = () => setCurrent((i) => (i - 1 + total) % total);
    const next = () => setCurrent((i) => (i + 1) % total);
    return (
      <div className="w-full flex justify-center items-center gap-6 mt-8 md:mt-10 z-20">
        <button
          aria-label="Previous image"
          onClick={prev}
          className="rounded-full border border-[var(--navy)] bg-white/70 text-[var(--navy)] w-8 h-8 flex items-center justify-center transition hover:bg-[var(--blue)] hover:text-white focus:outline-none focus:ring-2 focus:ring-[var(--blue)]"
        >
          <FaChevronLeft size={14} />
        </button>
        <div className="flex gap-2">
          {heroImages.map((_, idx) => (
            <button
              key={idx}
              aria-label={`Go to image ${idx + 1}`}
              onClick={() => setCurrent(idx)}
              className={`rounded-full w-4 h-4 flex items-center justify-center border border-transparent bg-white/60 transition ${current === idx ? "border-[var(--blue)]" : ""}`}
            >
              <span
                className={`block w-2 h-2 rounded-full ${current === idx ? "bg-[var(--blue)]" : "bg-[var(--navy)] opacity-30"}`}
              />
            </button>
          ))}
        </div>
        <button
          aria-label="Next image"
          onClick={next}
          className="rounded-full border border-[var(--navy)] bg-white/70 text-[var(--navy)] w-8 h-8 flex items-center justify-center transition hover:bg-[var(--blue)] hover:text-white focus:outline-none focus:ring-2 focus:ring-[var(--blue)]"
        >
          <FaChevronRight size={14} />
        </button>
      </div>
    );
  }

  return (
    <section>
      <section ref={sectionRef} className="relative overflow-hidden min-h-[65vh] md:min-h-[75vh]">
        <div className="absolute inset-0 -z-10">
          <Image
            src={heroImages[current]}
            alt="CUMSA banner"
            fill
            priority
            className="object-cover object-center transition-all duration-700"
          />
          <div className="absolute inset-0 hero-overlay" />
          <div className="hero-glow hero-glow-1" />
          <div className="hero-glow hero-glow-2" />
        </div>
        <div className="container-base py-16 md:py-28 text-white">
          <div className="max-w-3xl">
            <div className="frost-card p-6 md:p-8 animate-fade-in">
              <motion.h1
                initial={{ opacity: 0, y: 12 }}
                animate={{ opacity: 1, y: 0 }}
                transition={{ duration: 0.5, ease: "easeOut" }}
                className="text-2xl md:text-6xl font-extrabold leading-tight"
              >
                Carleton University Muslim Studentsâ€™ Association
              </motion.h1>
              <motion.div className="mt-3 md:mt-4 max-w-2xl text-white/90 text-lg">
                <Typewriter
                  options={{
                    autoStart: true,
                    loop: true,
                    delay: 40,
                    deleteSpeed: 25,
                    cursor: '<span class="typewriter-cursor">|</span>',
                  }}
                  onInit={(tw) => {
                    tw.typeString("Assalamu alaykum! Welcome to CUMSA.")
                      .pauseFor(4000)
                      .deleteAll()
                      .typeString("Find prayer times, events, and friendly faces.")
                      .pauseFor(4000)
                      .deleteAll()
                      .typeString("Family in Faith. You belong here.")
                      .pauseFor(4000)
                      .deleteAll()
                      .start();
                  }}
                />
              </motion.div>
              <motion.div
                initial={{ opacity: 0 }}
                animate={{ opacity: 1 }}
                transition={{ delay: 0.2, duration: 0.5 }}
                className="mt-6 md:mt-8 flex flex-wrap gap-3"
              >
                <Link
                  href="/services/prayer"
                  className="btn-cozy border-2 border-[color-mix(in_oklab,var(--blue)_40%,transparent)] bg-[color-mix(in_oklab,var(--blue)_10%,transparent)] text-white/90 hover:bg-[var(--blue)] hover:text-white transition-base"
                >
                  Prayers
                </Link>
                <Link
                  href="/events"
                  className="btn-cozy border-2 border-[color-mix(in_oklab,var(--blue)_40%,transparent)] bg-[color-mix(in_oklab,var(--blue)_10%,transparent)] text-white/90 hover:bg-[var(--blue)] hover:text-white transition-base"
                >
                  Events
                </Link>
              </motion.div>
            </div>
          </div>
        </div>
      </section>
      <SlideControls current={current} setCurrent={setCurrent} heroImages={heroImages} />
    </section>
  );
}
