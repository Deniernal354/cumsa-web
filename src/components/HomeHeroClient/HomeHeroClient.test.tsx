import { render } from "@testing-library/react";
import HomeHeroClient from "./HomeHeroClient";

test("HomeHeroClient snapshot", () => {
  const { asFragment } = render(<HomeHeroClient />);
  expect(asFragment()).toMatchSnapshot();
});
