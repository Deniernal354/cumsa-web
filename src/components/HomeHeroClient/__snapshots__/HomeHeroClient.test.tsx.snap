// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeHeroClient snapshot 1`] = `
<DocumentFragment>
  <section
    class="relative overflow-hidden"
  >
    <div
      class="absolute inset-0 -z-10"
    >
      <img
        alt="CUMSA banner"
        class="object-cover"
        src="/zahid-lilani-unsplash-scaled.jpg"
        style="position: absolute; inset: 0;"
      />
      <div
        class="absolute inset-0 bg-[var(--navy)]/55"
      />
    </div>
    <div
      class="container-base py-24 md:py-32 text-white"
    >
      <h1
        class="text-3xl md:text-5xl font-extrabold tracking-tight"
        style="opacity: 0; transform: translateY(12px);"
      >
        Carleton University Muslim Studentsâ€™ Association
      </h1>
      <div
        class="mt-4 max-w-2xl text-white/85 text-lg"
      >
        <div
          class="Typewriter"
          data-testid="typewriter-wrapper"
        >
          <span
            class="Typewriter__wrapper"
          />
          <span
            class="Typewriter__cursor"
          >
            <span
              class="typewriter-cursor"
            >
              |
            </span>
          </span>
        </div>
      </div>
      <div
        class="mt-8 flex flex-wrap gap-3"
        style="opacity: 0;"
      >
        <a
          class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none bg-[var(--blue)] text-white hover:opacity-90 focus-visible:ring-[var(--blue)] ring-offset-[var(--surface)] px-4 py-2"
          href="/services/prayer"
        >
          Prayers
        </a>
        <a
          class="inline-flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none border border-black/10 bg-[var(--surface)] text-[var(--foreground)] hover:bg-[var(--blue)] hover:text-white focus-visible:ring-[var(--blue)] ring-offset-[var(--surface)] px-4 py-2"
          href="/events"
        >
          Events
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
